crates/sui-core/src/authority/authority_store_tables.rs

```diff
@@ -239,61 +240,52 @@ impl ObjectStore for AuthorityPerpetualTables {
             .skip_prior_to(&ObjectKey::max_for_id(object_id))?
             .next();
 
-        let obj = match obj_entry {
-            Some((ObjectKey(obj_id, _), obj)) if obj_id == *object_id => obj,
-            _ => return Ok(None),
-        };
-
-        // Note that the two reads in this function are (obviously) not atomic, and the
-        // object may be deleted after we have read it. Hence we check get_latest_parent_entry
-        // last, so that the write to self.parent_sync gets the last word.
-        // Such race is ok because, even if the reads were atomic, calls to this function would still
-        // race with object deletions (the object could be deleted between when the function is
-        // called and when the first read takes place, which would be indistinguishable to the
-        // caller with the case in which the object is deleted in between the two reads).
-        let parent_entry = self.get_latest_parent_entry(*object_id)?;
-
-        match parent_entry {
-            None => {
-                error!(
-                    ?object_id,
-                    "Object is missing parent_sync entry, data store is inconsistent"
-                );
-                Ok(None)
-            }
-            Some((obj_ref, _)) if obj_ref.2.is_alive() => Ok(Some(self.object(obj)?)),
+        match obj_entry {
+            Some((ObjectKey(obj_id, _), obj)) if obj_id == *object_id => Ok(self.object(obj)?),
```

Comment:
is `self.object(obj)?` doing the is_alive check?

Reply:
yeah, sorry for confustion, I should add docstring to `object` function. It returns `Option<Object>` now. `None` means that objects doesn't exist anymore(or never existed). So alive check is not necessary anymore